server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -5793,4 +5804,67 @@ public AsyncShardRefreshTask getRefreshTask() {
         return refreshTask;
     }
 
+    public void startPeriodicFlushTask() {
+        TimeValue interval = indexSettings.getPeriodicFlushInterval();
+        // Only start the async flush task if interval is >0 and task is not already running
+        if (interval.millis() > 0 && periodicFlushTask == null) {
+            periodicFlushTask = new AsyncShardFlushTask(this, interval);
+            logger.info("Started periodic flush task for shard [{}] with interval [{}]", shardId, interval);
+        }
+    }
+
+    // Visible for testing
+    public AsyncShardFlushTask getPeriodicFlushTask() {
+        return periodicFlushTask;
+    }
+
+    /**
+     * Async shard flush task to call flush at a regular interval.
+     * This is particularly useful for pull-based ingestion index without translog to ensure offsets are regularly committed.
+     */
+    @ExperimentalApi
+    public final class AsyncShardFlushTask extends AbstractAsyncTask {
```

Comment:
I don't think this should be public. It's unnecessary API surface area. (I also don't think `AsyncShardRefreshTask` should be public, but that's another thing.)